sqlx/contract.go

```diff
@@ -15,18 +15,20 @@ import (
 type Tr interface {
 	sqlx.ExtContext
 
+	sqlx.Preparer
 	Preparex(query string) (*sqlx.Stmt, error)
 	PreparexContext(ctx context.Context, query string) (*sqlx.Stmt, error)
-	PrepareNamedContext(ctx context.Context, query string) (*sqlx.NamedStmt, error)
 	PrepareNamed(query string) (*sqlx.NamedStmt, error)
+	PrepareNamedContext(ctx context.Context, query string) (*sqlx.NamedStmt, error)
 
+	sqlx.Execer
 	MustExec(query string, args ...interface{}) sql.Result
 	MustExecContext(ctx context.Context, query string, args ...interface{}) sql.Result
-	NamedExecContext(ctx context.Context, query string, arg interface{}) (sql.Result, error)
 	NamedExec(query string, arg interface{}) (sql.Result, error)
+	NamedExecContext(ctx context.Context, query string, arg interface{}) (sql.Result, error)
 
-	Queryx(query string, args ...interface{}) (*sqlx.Rows, error)
-	QueryRowx(query string, args ...interface{}) *sqlx.Row
+	sqlx.Queryer
+	QueryRow(query string, args ...any) *sql.Row
```

Comment:
To solve that, the build tags can be used.

For example,
https://github.com/uber-go/multierr/blob/v1.10.0/error_post_go120.go
https://github.com/uber-go/multierr/blob/v1.10.0/error_pre_go120.go

Reply:
I changed the `any` type to `interface{},` and that fixed the build and tests